more_itertools/more.py

```diff
@@ -1121,11 +1137,27 @@ def sliced(seq, n):
         >>> list(sliced((1, 2, 3, 4, 5, 6, 7, 8), 3))
         [(1, 2, 3), (4, 5, 6), (7, 8)]
 
+    `sliced` raises a `ValueError` if *strict* is `True` and the length of
+    the sequence is not divisible by the requested slice length.
+
     This function will only work for iterables that support slicing.
     For non-sliceable iterables, see :func:`chunked`.
 
     """
-    return takewhile(len, (seq[i : i + n] for i in count(0, n)))
+    iterator = takewhile(len, (seq[i : i + n] for i in count(0, n)))
+    if strict:
+        # We return the wrapped iterator instead of doing a
+        # for/yield directly because otherwise the tests
+        # fail.
+        def ret():
+            for _slice in iterator:
+                if len(_slice) != n:
+                    raise ValueError("seq is not divisible by n.")
```

Comment:
@bbayles @shlomif 
Wouldn't this raise the exception only after you've consumed the entire iterator? Don't you think it should raise it when calling the function?